src/components/partner/partner.resolver.ts

```diff
@@ -93,6 +98,22 @@ export class PartnerResolver {
     );
   }
 
+  @ResolveField(() => SecuredFieldRegionList)
+  async fieldRegions(
+    @AnonSession() session: Session,
+    @Parent() partner: Partner,
+    @ListArg(FieldRegionListInput) input: FieldRegionListInput,
+    @Loader(FieldRegionLoader) loader: LoaderOf<FieldRegionLoader>,
+  ): Promise<SecuredFieldRegionList> {
+    const list = await this.partnerService.listFieldRegions(
+      partner,
+      input,
+      session,
+    );
+    loader.primeAll(list.items);
+    return list;
```

Comment:
Hmmm...okay.  I want to understand though.  I was following the example of the other list immediately below for `projects`.  Why the difference here?

Reply:
Well the main reason is, we expect this to be a short list (<25). So pagination/sorting/filtering is not needed (or can be done client side). Because of this we (do/should) eagerly grab the IDs of the field regions every time. This saves us and extra trip to the DB for this unique partner.
```gql
partner(id: "") {
  fieldRegions {
    id, name
  }
}
```
Two DB queries. One for partner, one for field regions.

```gql
partners {
  items {
    fieldRegions {
      id, name
    }
  }
}
```
1 + N queries (26 by default). One for partner list, one for each list of field regions.
But with the data loader suggestion above, it goes to 2. One for partner list, One for all the field regions needed.

Reply:
I mostly get that; thanks.